Fix link_inference activation function with 'ip' method

adocherty opened the pull request:
This bugfix applied the specified activation function when using the `edge_feature_method='ip'` option in the `link_inference` function.

This is particularly important as our example notebook uses the 'ip' method and specifies a 'sigmoid' activation.  The 'ip' method also produces the best generalization performance currently (perhaps because it has no trainable parameters).

The unit tests are also modified to test for 'linear' and 'sigmoid' activation functions.

adocherty commented on stellargraph/__init__.py:
```diff
@@ -3,8 +3,10 @@
 """
 
-__all__ = ["data", "layer", "mapper", "StellarDiGraph", "StellarGraph"]
+__all__ = ["data", "layer", "mapper", "StellarDiGraph", "StellarGraph", "__version__"]
```
There seems to be some conflicts between `black` and `clodeclimate` style!

TheDen replied:
🤔`black` has a section behind their rationale on line length limit https://github.com/ambv/black#line-length (set to `88`)

[pep8](https://www.python.org/dev/peps/pep-0008/#maximum-line-length) has the maximum line length as `79`, which is what codeclimate uses right now (through the `pep8` plugin).

Going by the list of [codeclimate plugins](https://docs.codeclimate.com/docs/list-of-engines), I can try disabling the `pep8` plugin, and try `sonarpython` + `Radon` + `Bandit` (checks for sec issues), and see how it works out.

adocherty replied:
I agree with black on that one. What does `radon` give over `sonarpython`? They both seem to do complexity checking.

youph commented on stellargraph/version.py:
```diff
@@ -0,0 +1,2 @@
+# Global version information
```
Good simplification!

adocherty commented:
There is a puzzle here. I get around 72% accuracy using the fix with a sigmoid activation function. However, I get around 82% accuracy using the original model, by explicitly setting the activation function to "linear".

Some notes on this:
* The binary_cross_entropy loss is calculated by clipping the input values into the range (0,1) and calculating the loss, so negative numbers and numbers over 1 are not a problem. In fact we are really using this as an effective nonlinearity.
* The embeddings are normalized by default so the inner product predictions are well scaled.
* Using the sigmoid nonlinearity, the predictions are in the limited range of (0.28, 0.73) this could be due to the default L2 normalization of the embeddings in the graphsage model.

Setting `normalization='none'` in the graphsage model gives the following:
* There was another bug that I introduced when allowing the normalization function to be specified: it reset the normalization to 'l2' irrespective of the setting. I fixed this in the latest commit.
* With the sigmoid nonlinearity we now attain around 72% accuracy
* With the linear function we now attain 70% accuracy

So, it seems that the normalization of the embeddings affects the direct use of a sigmoid nonlinearity over the inner product. However, even when using no normalization the sigmoid still performs worse than the direct linear.

For now I'm specifying a 'relu' function as the nonlinearity – at least this means that the output of the inner product will be positive, and due to the clipping in the `binary_cross_entropy` function this is implicitly what we have been using!

We should discuss what to do about this. Using linear outputs directly seems to work better, but this seems like it's bad practice in terms of machine learning. Another thing to try is a dense layer over the IP outputs: two trainable parameters that learn the scale and offset for the scalar value of the IP output before feeding to the sigmoid function.
